src/components/QuickOpenModal.js

```diff
@@ -68,10 +70,18 @@ type GotoLocationType = {
 };
 
 function filter(values, query) {
-  return fuzzyAldrin.filter(values, query, {
+  let results = fuzzyAldrin.filter(values, query, {
     key: "value",
     maxResults: 1000
   });
+
+  if (prefs.projectDirectoryRoot) {
+    results = results.filter(result =>
+      result.url.includes(prefs.projectDirectoryRoot)
+    );
+  }
+
+  return results;
```

Comment:
this is awesome!

I think we should also update:

`formatSources(getSources(state), prefs.projectDirectoryRoot)`  so that we can truncate the paths, which will really clean up the UI!

Reply:
Good shout -- added as a "to do" and marked the PR WIP